### PR TITLE
Mekhanik evgenii/fix 1901 1

### PR DESCRIPTION
**Pull request by EvgeniiMekhanik**

### CODE REVIEW

**Review by const-t — APPROVED** [review]
LGTM

File: fw/http_parser.c [hunk 1]
```diff
@@ -7532,6 +7605,38 @@ __h2_req_parse_content_length(TfwHttpMsg *msg, unsigned char *data, size_t len,
 	return ret >= 0 ? CSTR_NEQ : ret;
 }
 
+int
+h2_set_hdr_content_type(TfwHttpReq *req, const TfwCachedHeaderState *cstate)
+{
+	int r;
+
+	if (cstate->is_set) {
```
Comment (krizhanovsky) [inline fw/http_parser.c:7613]: IIRC the Linux kernel coding style also suggests early returns:

```suggestion
	if (!cstate->is_set)
		return 0;
```
and probably we can avoid some line breaks.

File: fw/http_parser.c [hunk 2]
```diff
@@ -7532,6 +7605,38 @@ __h2_req_parse_content_length(TfwHttpMsg *msg, unsigned char *data, size_t len,
...
+			cstate->content_type.multipart_boundary_raw.len;
+		req->multipart_boundary.len =
+			cstate->content_type.multipart_boundary.len;
+		req->multipart_boundary_raw.nchunks =
+			cstate->content_type.multipart_boundary_raw.nchunks;
+		if(test_bit(TFW_HTTP_B_CT_MULTIPART,
+			    cstate->content_type.flags)) {
+			__set_bit(TFW_HTTP_B_CT_MULTIPART, req->flags);
+		}
+		if(test_bit(TFW_HTTP_B_CT_MULTIPART_HAS_BOUNDARY,
```
Comment (krizhanovsky) [inline fw/http_parser.c:7624]: here and above: we place spaces between operators like `if` and (

File: fw/http_parser.c [hunk 3]
```diff
@@ -8415,25 +8548,40 @@ __h2_req_parse_if_msince(TfwHttpMsg *msg, unsigned char *data, size_t len,
...
 	}
 
 	return r;
 }
 
+void
+h2_set_hdr_pragma(TfwHttpReq *req, const TfwCachedHeaderState *cstate)
+{
+	if (cstate->is_set &&
+	    cstate->cache_ctl.flags & TFW_HTTP_CC_PRAGMA_NO_CACHE) {
```
Comment (krizhanovsky) [inline fw/http_parser.c:8561]: it'd better to place { on a new line to make the multi-line condition more readable

File: fw/http_parser.c [hunk 4]
```diff
@@ -8415,25 +8548,40 @@ __h2_req_parse_if_msince(TfwHttpMsg *msg, unsigned char *data, size_t len,
...
 __h2_req_parse_pragma(TfwHttpMsg *hm, unsigned char *data, size_t len, bool fin)
 {
 	int r = CSTR_NEQ;
 	__FSM_DECLARE_VARS(hm);
 
+#define __SET_CACHE_CTL_FLAG(flag)					\
+do {									\
+	msg->cache_ctl.flags |= flag;					\
+	parser->cstate.cache_ctl.flags |= flag;				\
+	parser->cstate.is_set = true;				\
```
Comment (krizhanovsky) [inline fw/http_parser.c:8577]: bad alignement of \

**Review by krizhanovsky — APPROVED** [review]
LGTM, just please make the couple of cleanups

**Pull request merged**